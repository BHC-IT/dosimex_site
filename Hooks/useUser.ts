import { useState, useEffect } from 'react';

export interface IUser {
	username: string,
	password: string,
}


const useUser = () => {

	const [token, setToken] = useState<string | null>(null)

	useEffect(() => {
		setToken(localStorage.getItem('access_token'))
	}, []);

	return token;
}

export default useUser;
