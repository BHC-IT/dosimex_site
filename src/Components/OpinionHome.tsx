import * as React from 'react';
import Radium from 'radium';
import * as CSS from 'csstype';

export interface IStyles {
	button: CSS.Properties,
	global: CSS.Properties,
	icon: CSS.Properties,
	peopleInfo: CSS.Properties,
}

interface IProps {
	text: any,
}

const OpinionHome = (props: IProps) => {

	return (
		<div style={styles.global}>
			<div style={styles.peopleInfo}>
				<h5 style={{fontSize: "1.8rem", margin: "0"}}>{props.text.name}</h5>
				<p style={{margin: "0"}}>{props.text.job}</p>
			</div>
			<div style={{margin: "4vh 0"}}>
				<img src="/Images/icon_quote.png" alt="icône citation" />
				<p style={{textAlign: "justify"}}>{props.text.opinion}</p>
			</div>
			<a
				style={styles.button}
				href="https://www.dosimex.fr/static/media/extrait_retour_utilisateurs.73178914.pdf"
				target="_blank">Tous les avis
				<img style={styles.icon} src="/Images/icon_download.png" alt="icône télécharger" />
			</a>
		</div>
	);

}

export default Radium(OpinionHome);

export const styles =  {
	button: {
		padding: "8px 25px",
		border: "2px solid var(--main)",
		borderRadius: "50px",
		color: "var(--main)",
		textTransform: "uppercase",
		transition: "all 0.3s ease 0s",
		':hover': {
			transform: "translateY(-4px)",
			boxShadow: "0px 5px 5px rgba(0, 0, 0, 0.1)",
		},
		display: "flex",
		alignItems: "center",
		justifyContent: "space-around",
		width: "220px",
		marginLeft: "auto",
		marginRight: "auto",
		marginTop: "7vh",
		marginBottom: "2vh",
	},
	global: {
		backgroundColor: "var(--grey-bg)",
		padding: "10vh 10vw",
	},
	icon: {
		width: "1.2vw",
	}
}