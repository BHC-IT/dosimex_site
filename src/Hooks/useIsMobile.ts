import * as React from 'react'

import { isMobile, isTablet } from 'react-device-detect'

export const useIsMobile = <T>(styles: (m: boolean) => T): T | null => {
	// @ts-ignore - React hooks compatibility with React 18
	const [style, setStyle] = React.useState<T | null>(styles(isMobile))

	// @ts-ignore - React hooks compatibility with React 18
	React.useEffect(() => {
		setStyle(null)
		setTimeout(() => setStyle(styles(isMobile)), 10)
	}, [])

	return style
}

export const useMobile = (): boolean | null => {
	// @ts-ignore - React hooks compatibility with React 18
	const [mobile, setMobile] = React.useState<boolean | null>(null)

	// @ts-ignore - React hooks compatibility with React 18
	React.useEffect(() => {
		setTimeout(() => setMobile(isMobile), 10)
	}, [])

	return mobile
}

export const useTablet = (): boolean | null => {
	// @ts-ignore - React hooks compatibility with React 18
	const [tablet, setTablet] = React.useState<boolean | null>(null)

	// @ts-ignore - React hooks compatibility with React 18
	React.useEffect(() => {
		setTimeout(() => setTablet(isTablet), 10)
	}, [])

	return tablet
}

export const useDeviceType = (): { isMobile: boolean | null; isTablet: boolean | null } => {
	const mobile = useMobile()
	const tablet = useTablet()

	return { isMobile: mobile, isTablet: tablet }
}
