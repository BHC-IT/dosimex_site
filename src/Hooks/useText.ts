import { useRouter } from 'next/router';

import ILang from '../lang/interface';

import fr from '../lang/fr';
import en from '../lang/en';
import debug from '../lang/debug';

export const text : {[$:string]: ILang} = {fr, en, debug};

export const useText = (page: string) : any => {
	const route = useRouter();

	if (route.locale === 'debug')
		return text[route.locale][page];
	else if (route?.locale)
		return text[route.locale.slice(0,2)]?.[page];
	return null
}
