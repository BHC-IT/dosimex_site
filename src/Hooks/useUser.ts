import { useState, useEffect } from 'react';


export interface IUser {
	username: string,
	password: string,
}

const remove = (setToken: Function) => {
	localStorage.removeItem('access_token')
	localStorage.removeItem('exp_token')
	setToken(null)
}

const useUser = () => {

	const [token, setToken] = useState<string | null>(null)

	useEffect(() => {
		setToken(localStorage.getItem('access_token'))
		const exp = Number(localStorage.getItem('exp_token'))

		if (!exp) {
			remove(setToken)
			return;
		}
		if (exp < Date.now())
			remove(setToken)
		else
			setTimeout(() => remove(setToken), exp - Date.now())
	}, []);

	return token;
}

export default useUser;
