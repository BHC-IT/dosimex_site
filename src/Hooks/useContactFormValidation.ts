import emailjs from '@emailjs/browser'
import { useState, useCallback } from 'react'
import { isPossiblePhoneNumber } from 'react-phone-number-input'
import { toast } from 'react-toastify'

import { IValidator } from '../Components/Input'

// Email validation regex
const EMAIL_REGEX =
	/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/

// Validation functions
export const isInputValid = (value: string): boolean => value.trim() !== ''
export const isEmailValid = (value: string): boolean => EMAIL_REGEX.test(value)
export const isPhoneValidNumber = (phone: string | undefined): boolean =>
	!phone || isPossiblePhoneNumber(phone)

// Validator factory functions
export const createRequiredValidator = (errorMessage: string): IValidator => ({
	validationFunction: value => isInputValid(value),
	errorMessage,
})

export const createEmailValidator = (errorMessage: string): IValidator => ({
	validationFunction: value => isEmailValid(value),
	errorMessage,
})

export interface IContactFormValidation {
	// Form field states
	name: string | null
	email: string | null
	phone: string | undefined
	enterprise: string | null
	address: string | null
	subject: string | null
	message: string | null

	// Validation states
	nameValid: boolean
	phoneError: string | null
	isLoading: boolean

	// Form handlers
	setName: (value: string) => void
	setEmail: (value: string) => void
	setPhone: (value: string | undefined) => void
	setEnterprise: (value: string) => void
	setAddress: (value: string) => void
	setSubject: (value: string) => void
	setMessage: (value: string) => void
	setNameValid: (valid: boolean) => void
	setIsLoading: (loading: boolean) => void

	// Validation methods
	validatePhone: () => void
	validateForm: () => boolean
	resetForm: () => void
	handleSubmit: (e: React.FormEvent) => void

	// Validator configurations
	getNameValidators: (errorMessage: string) => IValidator[]
	getEmailValidators: (errorEmail: string[]) => IValidator[]
	getAddressValidators: (errorMessage: string) => IValidator[]
}

// Validation helper functions
const validateName = (name: string | null): boolean => {
	return name !== null && name.trim() !== ''
}

const validateEmail = (email: string | null): boolean => {
	if (!email || email.trim() === '') return false
	const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
	return emailRegex.test(email)
}

const validateMessage = (message: string | null): boolean => {
	return message !== null && message.trim() !== ''
}

// Custom hook for form state management
const useFormState = () => {
	const [name, setName] = useState<string | null>(null)
	const [email, setEmail] = useState<string | null>(null)
	const [phone, setPhone] = useState<string | undefined>(undefined)
	const [enterprise, setEnterprise] = useState<string | null>(null)
	const [address, setAddress] = useState<string | null>(null)
	const [subject, setSubject] = useState<string | null>(null)
	const [message, setMessage] = useState<string | null>(null)
	const [nameValid, setNameValid] = useState(false)
	const [isLoading, setIsLoading] = useState(false)

	return {
		name, setName,
		email, setEmail,
		phone, setPhone,
		enterprise, setEnterprise,
		address, setAddress,
		subject, setSubject,
		message, setMessage,
		nameValid, setNameValid,
		isLoading, setIsLoading,
	}
}

// Form validation logic
const useFormValidation = (nameValid: boolean, phone: string | undefined) => {
	const checkInputs = useCallback((): boolean => {
		return nameValid && (phone ? isPossiblePhoneNumber(phone) : true)
	}, [nameValid, phone])

	const validateForm = useCallback(() => checkInputs(), [checkInputs])

	const validatePhone = useCallback(() => {
		// Phone validation is done in checkInputs
	}, [])

	return { checkInputs, validateForm, validatePhone }
}

// Form reset functionality
const useFormReset = (
	setName: (value: string | null) => void,
	setEmail: (value: string | null) => void,
	setPhone: (value: string | undefined) => void,
	setEnterprise: (value: string | null) => void,
	setAddress: (value: string | null) => void,
	setSubject: (value: string | null) => void,
	setMessage: (value: string | null) => void,
	setNameValid: (valid: boolean) => void,
	setIsLoading: (loading: boolean) => void,
) => {
	const resetForm = useCallback(() => {
		setName(null)
		setEmail(null)
		setPhone(undefined)
		setEnterprise(null)
		setAddress(null)
		setSubject(null)
		setMessage(null)
		setNameValid(false)
		setIsLoading(false)
	}, [setName, setEmail, setPhone, setEnterprise, setAddress, setSubject, setMessage, setNameValid, setIsLoading])

	return { resetForm }
}

// Email sending functionality
const useEmailSending = (
	name: string | null,
	email: string | null,
	phone: string | undefined,
	enterprise: string | null,
	address: string | null,
	subject: string | null,
	message: string | null,
	text: Record<string, string | string[]> | undefined,
	setIsLoading: (loading: boolean) => void,
	resetForm: () => void,
) => {
	const sendEmail = useCallback((): void => {
		sendContactEmail({
			name, email, phone, enterprise, address, subject, message,
			text, setIsLoading, resetForm,
		})
	}, [name, email, phone, enterprise, address, subject, message, text, setIsLoading, resetForm])

	return { sendEmail }
}

// Validator creation functions
const useValidatorCreation = (
	name: string | null,
	email: string | null,
	message: string | null,
) => {
	const getNameValidators = useCallback((errorName: string): IValidator[] => [
		{
			validationFunction: () => validateName(name),
			errorMessage: errorName || 'Name is required',
		},
	], [name])

	const getEmailValidators = useCallback((errorEmail: string[]): IValidator[] => [
		{
			validationFunction: () => email !== null && email.trim() !== '',
			errorMessage: errorEmail[0] || 'Email is required',
		},
		{
			validationFunction: () => validateEmail(email),
			errorMessage: errorEmail[1] || 'Email format is invalid',
		},
	], [email])

	const getAddressValidators = useCallback((errorMessage: string): IValidator[] => [
		{
			validationFunction: () => validateMessage(message),
			errorMessage: errorMessage || 'Message is required',
		},
	], [message])

	return { getNameValidators, getEmailValidators, getAddressValidators }
}

// Form submission handling
const useFormSubmission = (checkInputs: () => boolean, sendEmail: () => void) => {
	const handleSubmit = useCallback((e: React.FormEvent) => {
		e.preventDefault()
		if (checkInputs()) {
			sendEmail()
		}
	}, [checkInputs, sendEmail])

	return { handleSubmit }
}

export const useContactFormValidation = (text?: Record<string, string | string[]>): IContactFormValidation => {
	const formState = useFormState()
	const formValidation = useFormValidation(formState.nameValid, formState.phone)
	const { resetForm } = useFormReset(
		formState.setName,
		formState.setEmail,
		formState.setPhone,
		formState.setEnterprise,
		formState.setAddress,
		formState.setSubject,
		formState.setMessage,
		formState.setNameValid,
		formState.setIsLoading,
	)
	const { sendEmail } = useEmailSending(
		formState.name,
		formState.email,
		formState.phone,
		formState.enterprise,
		formState.address,
		formState.subject,
		formState.message,
		text,
		formState.setIsLoading,
		resetForm,
	)
	const validatorCreation = useValidatorCreation(formState.name, formState.email, formState.message)
	const { handleSubmit } = useFormSubmission(formValidation.checkInputs, sendEmail)

	return {
		// Form field states
		name: formState.name,
		email: formState.email,
		phone: formState.phone,
		enterprise: formState.enterprise,
		address: formState.address,
		subject: formState.subject,
		message: formState.message,

		// Validation states
		nameValid: formState.nameValid,
		phoneError: null, // No longer tracking phone errors separately
		isLoading: formState.isLoading,

		// Form handlers
		setName: formState.setName,
		setEmail: formState.setEmail,
		setPhone: formState.setPhone,
		setEnterprise: formState.setEnterprise,
		setAddress: formState.setAddress,
		setSubject: formState.setSubject,
		setMessage: formState.setMessage,
		setNameValid: formState.setNameValid,
		setIsLoading: formState.setIsLoading,

		// Validation methods
		validatePhone: formValidation.validatePhone,
		validateForm: formValidation.validateForm,
		resetForm,
		handleSubmit,

		// Validator configurations
		getNameValidators: validatorCreation.getNameValidators,
		getEmailValidators: validatorCreation.getEmailValidators,
		getAddressValidators: validatorCreation.getAddressValidators,
	}
}

// Email sending helper function
const sendContactEmail = (params: {
	name: string | null, email: string | null, phone: string | undefined,
	enterprise: string | null, address: string | null, subject: string | null,
	message: string | null, text: Record<string, string | string[]> | undefined,
	setIsLoading: (loading: boolean) => void,
	resetForm: () => void
}) => {
	const { name, email, phone, enterprise, address, subject, message, text, setIsLoading, resetForm } = params
	const toastLoad = toast.dark(text?.sending)
	setIsLoading(true)
	emailjs
		.send(
			process.env.NEXT_PUBLIC_EMAILJS_SERVICE_ID ?? '',
			process.env.NEXT_PUBLIC_EMAILJS_TEMPLATE_ID ?? '',
			{
					text: message,
					name,
					email,
					phone,
					enterprise,
					address,
					subject,
				},
				process.env.NEXT_PUBLIC_EMAILJS_PUBLIC_KEY ?? '',
			)
			.then(() => {
				toast.dismiss(toastLoad)
				toast.success(text?.messageSent)
				resetForm()
			})
			.catch(() => {
				toast.dismiss(toastLoad)
				toast.error(text?.messageNotSent)
			})
			.finally(() => {
				setIsLoading(false)
			})
}
