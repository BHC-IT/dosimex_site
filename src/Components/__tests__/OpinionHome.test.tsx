import { render, screen, fireEvent } from '@testing-library/react'
import { describe, expect, it, vi, beforeEach } from 'vitest'

import { useIsMobile } from '../../Hooks/useIsMobile'
import OpinionHome, { styles } from '../OpinionHome'

// Mock the useIsMobile hook
vi.mock('../../Hooks/useIsMobile', () => ({
	useIsMobile: vi.fn(),
}))

// Test data factory
const createMockText = (overrides = {}) => ({
	name: 'John Doe',
	job: 'Software Engineer',
	opinion: 'Great product! Very easy to use and efficient.',
	btn: 'Download PDF',
	...overrides,
})

const createMockStyle = (mobile = false) => styles(mobile)

describe('OpinionHome Component', () => {
	beforeEach(() => {
		vi.mocked(useIsMobile).mockReturnValue(createMockStyle(false))
	})

	describe('Rendering', () => {
		it('should render correctly with text', () => {
			const mockText = createMockText()
			const { container } = render(<OpinionHome text={mockText} />)
			expect(container.firstChild).toMatchSnapshot()
		})

		it('should render component structure', () => {
			const mockText = createMockText()
			const { container } = render(<OpinionHome text={mockText} />)

			// Component should render without errors
			expect(container).toBeDefined()
			expect(container.firstChild).not.toBeNull()
		})

		it('should return null when style is null (SSR protection)', () => {
			vi.mocked(useIsMobile).mockReturnValue(null)
			const mockText = createMockText()
			const { container } = render(<OpinionHome text={mockText} />)

			expect(container.firstChild).toBeNull()
		})
	})

	describe('Content Display', () => {
		it('should display person information correctly', () => {
			const mockText = createMockText({
				name: 'Jane Smith',
				job: 'Product Manager',
				opinion: 'Excellent tool for our team!',
				btn: 'Get Report',
			})

			render(<OpinionHome text={mockText} />)

			expect(screen.getByText('Jane Smith')).toBeInTheDocument()
			expect(screen.getByText('Product Manager')).toBeInTheDocument()
			expect(screen.getByText('Excellent tool for our team!')).toBeInTheDocument()
			expect(screen.getByText('Get Report')).toBeInTheDocument()
		})

		it('should display quote icon with correct alt text', () => {
			const mockText = createMockText()
			render(<OpinionHome text={mockText} />)

			const quoteIcon = screen.getByAltText('icône citation')
			expect(quoteIcon).toBeInTheDocument()
			expect(quoteIcon).toHaveAttribute('src', '/Images/icon_quote.png')
		})

		it('should display download icon with correct alt text', () => {
			const mockText = createMockText()
			render(<OpinionHome text={mockText} />)

			const downloadIcon = screen.getByAltText('icône télécharger')
			expect(downloadIcon).toBeInTheDocument()
			expect(downloadIcon).toHaveAttribute('src', '/Images/icon_download.png')
		})
	})

	describe('Download Link Functionality', () => {
		it('should render download link with correct attributes', () => {
			const mockText = createMockText()
			render(<OpinionHome text={mockText} />)

			const downloadLink = screen.getByRole('link', { name: /download pdf/i })
			expect(downloadLink).toBeInTheDocument()
			expect(downloadLink).toHaveAttribute('href', '../Folders/extrait_retour_utilisateurs.pdf')
			expect(downloadLink).toHaveAttribute('target', '_blank')
			expect(downloadLink).toHaveAttribute('rel', 'noreferrer noopener')
		})

		it('should be accessible to screen readers', () => {
			const mockText = createMockText()
			render(<OpinionHome text={mockText} />)

			const downloadLink = screen.getByRole('link')
			expect(downloadLink).toBeInTheDocument()

			// The link should contain both text and icon for accessibility
			expect(downloadLink).toHaveTextContent('Download PDF')
			const icon = downloadLink.querySelector('img[alt="icône télécharger"]')
			expect(icon).toBeInTheDocument()
		})
	})

	describe('Hover Interactions', () => {
		it('should handle mouse enter and leave events', () => {
			const mockText = createMockText()
			render(<OpinionHome text={mockText} />)

			const downloadLink = screen.getByRole('link')

			// Initial state (no hover)
			expect(downloadLink).toBeInTheDocument()

			// Mouse enter
			fireEvent.mouseEnter(downloadLink)
			expect(downloadLink).toBeInTheDocument()

			// Mouse leave
			fireEvent.mouseLeave(downloadLink)
			expect(downloadLink).toBeInTheDocument()
		})

		it('should apply hover styles when hovered', () => {
			const mockText = createMockText()
			render(<OpinionHome text={mockText} />)

			const downloadLink = screen.getByRole('link')

			// Check initial style (base styles applied)
			expect(downloadLink).toHaveStyle({
				padding: '8px 25px',
				borderRadius: '50px',
				textTransform: 'uppercase',
			})

			// Trigger hover
			fireEvent.mouseEnter(downloadLink)

			// The component should handle hover state internally
			// We can't easily test the dynamic style changes without more complex setup
			expect(downloadLink).toBeInTheDocument()
		})
	})

	describe('Responsive Behavior', () => {
		it('should use mobile styles when mobile is true', () => {
			vi.mocked(useIsMobile).mockReturnValue(createMockStyle(true))
			const mockText = createMockText()

			render(<OpinionHome text={mockText} />)

			// Component should render with mobile styles
			const downloadIcon = screen.getByAltText('icône télécharger')
			expect(downloadIcon).toBeInTheDocument()
		})

		it('should use desktop styles when mobile is false', () => {
			vi.mocked(useIsMobile).mockReturnValue(createMockStyle(false))
			const mockText = createMockText()

			render(<OpinionHome text={mockText} />)

			// Component should render with desktop styles
			const downloadIcon = screen.getByAltText('icône télécharger')
			expect(downloadIcon).toBeInTheDocument()
		})
	})

	describe('Edge Cases', () => {
		it('should handle empty text properties gracefully', () => {
			const mockText = createMockText({
				name: '',
				job: '',
				opinion: '',
				btn: '',
			})

			render(<OpinionHome text={mockText} />)

			// Component should render even with empty text
			const quoteIcon = screen.getByAltText('icône citation')
			expect(quoteIcon).toBeInTheDocument()
		})

		it('should handle long text content', () => {
			const mockText = createMockText({
				name: 'Dr. Elizabeth Johnson-Williams-Smith',
				job: 'Senior Lead Principal Software Engineering Architect',
				opinion: 'This is an extremely long opinion text that goes on and on about how amazing this product is. It includes many details about the user experience, performance benefits, ease of use, and overall satisfaction with the tool. The text continues to provide extensive feedback about various features and capabilities.',
				btn: 'Download Complete Report',
			})

			render(<OpinionHome text={mockText} />)

			expect(screen.getByText('Dr. Elizabeth Johnson-Williams-Smith')).toBeInTheDocument()
			expect(screen.getByText('Senior Lead Principal Software Engineering Architect')).toBeInTheDocument()
			expect(screen.getByText(/This is an extremely long opinion text/)).toBeInTheDocument()
			expect(screen.getByText('Download Complete Report')).toBeInTheDocument()
		})
	})
})

describe('OpinionHome Styles', () => {
	describe('Desktop Styles', () => {
		it('should have correct base button styles for desktop', () => {
			const desktopStyles = styles(false)

			expect(desktopStyles.button.base).toEqual({
				padding: '8px 25px',
				border: '2px solid var(--main)',
				borderRadius: '50px',
				color: 'var(--main)',
				textTransform: 'uppercase',
				transition: 'all 0.3s ease 0s',
				display: 'flex',
				alignItems: 'center',
				justifyContent: 'space-around',
				width: '220px',
				marginLeft: 'auto',
				marginRight: 'auto',
				marginTop: '7vh',
				marginBottom: '2vh',
				textDecoration: 'none',
			})
		})

		it('should have correct hover styles', () => {
			const desktopStyles = styles(false)

			expect(desktopStyles.button.hover).toEqual({
				transform: 'translateY(-4px)',
				boxShadow: '0px 5px 5px rgba(0, 0, 0, 0.1)',
			})
		})

		it('should have correct icon styles for desktop', () => {
			const desktopStyles = styles(false)

			expect(desktopStyles.icon).toEqual({
				width: '1.2vw',
				minWidth: '25px',
				maxWidth: undefined,
			})
		})
	})

	describe('Mobile Styles', () => {
		it('should have correct icon styles for mobile', () => {
			const mobileStyles = styles(true)

			expect(mobileStyles.icon).toEqual({
				width: '4.7vw',
				minWidth: undefined,
				maxWidth: '25px',
			})
		})

		it('should have correct text styles for mobile', () => {
			const mobileStyles = styles(true)

			expect(mobileStyles.text).toEqual({
				textAlign: 'justify',
				fontSize: '1.5rem',
			})
		})
	})

	describe('Common Styles', () => {
		it('should have correct global container styles', () => {
			const styles_desktop = styles(false)
			const styles_mobile = styles(true)

			// Both desktop and mobile should have same global styles
			const expectedGlobalStyles = {
				backgroundColor: 'var(--grey-bg)',
				padding: '10vh 10vw',
			}

			expect(styles_desktop.global).toEqual(expectedGlobalStyles)
			expect(styles_mobile.global).toEqual(expectedGlobalStyles)
		})

		it('should have correct name heading styles', () => {
			const desktopStyles = styles(false)

			expect(desktopStyles.nameHeading).toEqual({
				fontSize: '1.8rem',
				margin: '0',
			})
		})

		it('should have correct job text styles', () => {
			const desktopStyles = styles(false)

			expect(desktopStyles.jobText).toEqual({
				margin: '0',
			})
		})

		it('should have correct quote container styles', () => {
			const desktopStyles = styles(false)

			expect(desktopStyles.quoteContainer).toEqual({
				margin: '4vh 0',
			})
		})
	})
})
