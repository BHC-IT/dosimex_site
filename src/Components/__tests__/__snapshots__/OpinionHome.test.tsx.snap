// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OpinionHome Component > Rendering > should render correctly with text 1`] = `
<div
  style="background-color: var(--grey-bg); padding: 10vh 10vw;"
>
  <div>
    <h5
      style="font-size: 1.8rem; margin: 0px;"
    >
      John Doe
    </h5>
    <p
      style="margin: 0px;"
    >
      Software Engineer
    </p>
  </div>
  <div
    style="margin: 4vh 0px;"
  >
    <img
      alt="icône citation"
      src="/Images/icon_quote.png"
    />
    <p
      style="text-align: justify;"
    >
      Great product! Very easy to use and efficient.
    </p>
  </div>
  <a
    href="../Folders/extrait_retour_utilisateurs.pdf"
    rel="noreferrer noopener"
    style="padding: 8px 25px; border: 2px solid var(--main); border-radius: 50px; color: var(--main); text-transform: uppercase; transition: all 0.3s ease 0s; display: flex; align-items: center; justify-content: space-around; width: 220px; margin: 7vh auto 2vh auto; text-decoration: none;"
    target="_blank"
  >
    Download PDF
    <img
      alt="icône télécharger"
      src="/Images/icon_download.png"
      style="width: 1.2vw; min-width: 25px;"
    />
  </a>
</div>
`;

exports[`OpinionHome Component > should render correctly with text 1`] = `null`;
